Marlin/temperature.cpp

```diff
@@ -376,9 +420,13 @@ void PID_autotune(float temp, int extruder, int ncycles)
 
 void updatePID() {
   #ifdef PIDTEMP
-    for (int e = 0; e < EXTRUDERS; e++) {
-      temp_iState_max[e] = PID_INTEGRAL_DRIVE_MAX / PID_PARAM(Ki,e);
-    }
+    #ifndef SINGLENOZZLE
+      for (int e = 0; e < EXTRUDERS; e++) {
```

Comment:
And also use `NOZZLES` here for the loop (and the compiler will optimize the loop out).